go/tools/builders/compilepkg.go

```diff
@@ -517,44 +522,25 @@ func compileArchive(
 	// Pack .o files into the archive. These may come from cgo generated code,
 	// cgo dependencies (cdeps), or assembly.
 	if len(objFiles) > 0 {
-		if err := appendFiles(goenv, outPath, objFiles); err != nil {
+		if err := appendToArchive(goenv, outPath, objFiles); err != nil {
 			return err
 		}
 	}
 
 	// Check results from nogo.
-	nogoStatus := nogoNotRun
 	if nogoChan != nil {
 		err := <-nogoChan
 		nogoChan = nil // no cancellation needed
 		if err != nil {
-			nogoStatus = nogoFailed
-			// TODO: should we still create the .x file without nogo facts in this case?
+			// TODO: Move nogo into a separate action so we don't fail the compilation here.
 			return err
 		}
-		nogoStatus = nogoSucceeded
-	}
-
-	// Extract the export data file and pack it in an .x archive together with the
-	// nogo facts file (if there is one). This allows compile actions to depend
-	// on .x files only, so we don't need to recompile a package when one of its
-	// imports changes in a way that doesn't affect export data.
-	// TODO(golang/go#33820): After Go 1.16 is the minimum supported version,
-	// use -linkobj to tell the compiler to create separate .a and .x files for
-	// compiled code and export data. Before that version, the linker needed
-	// export data in the .a file when building a plugin. To work around that,
-	// we copy the export data into .x ourselves.
-	if err = extractFileFromArchive(outPath, workDir, pkgDef); err != nil {
-		return err
-	}
-	pkgDefPath := filepath.Join(workDir, pkgDef)
-	if nogoStatus == nogoSucceeded {
-		return appendFiles(goenv, outXPath, []string{pkgDefPath, outFactsPath})
 	}
-	return appendFiles(goenv, outXPath, []string{pkgDefPath})
+
+	return nil
```

Comment:
could you clarify whether if the old TODO is still relevant?
Afaik we are not using linkobj flag anywhere. Was it replaced by something else?

Reply:
oh nvm I found the flag a few files below 😮‍💨